metomi/rose/config_processors/fileinstall.py

```diff
@@ -253,6 +253,19 @@ def _process(self, conf_tree, nodes, loc_dao, **kwargs):
                         if prev_path != path:
                             target.is_out_of_date = True
                             break
+                # See if any sources have changed names.
+                if not target.is_out_of_date:
+                    conn = loc_dao.get_conn()
+                    prev_dep_locs = conn.execute(
+                        "SELECT * FROM dep_names WHERE name=?", [target.name]
+                    ).fetchall()
+                    prev_dep_locs = [i[1] for i in prev_dep_locs]
+                    prev_dep_locs = [loc_dao.select(i) for i in prev_dep_locs]
+                    if (
+                        sorted([i.name for i in prev_dep_locs]) !=
+                        sorted([i.name for i in target.dep_locs])
+                    ):
+                        target.is_out_of_date = True
```

Comment:
I think we should wrap that with a `conn.close` for safety (unless it's done somewhere else?):

```suggestion
                    try:
                        prev_dep_locs = conn.execute(
                            "SELECT * FROM dep_names WHERE name=?", [target.name]
                        ).fetchall()
                        prev_dep_locs = [i[1] for i in prev_dep_locs]
                        prev_dep_locs = [loc_dao.select(i) for i in prev_dep_locs]
                        if (
                            sorted([i.name for i in prev_dep_locs]) !=
                            sorted([i.name for i in target.dep_locs])
                        ):
                            target.is_out_of_date = True
                    finally:
                        conn.close()
```